src/com/google/javascript/jscomp/AstAnalyzer.java

```diff
@@ -520,6 +524,7 @@ boolean nodeTypeMayHaveSideEffects(Node n) {
       case FOR_IN: // assigns to a loop LHS
       case FOR_OF: // assigns to a loop LHS, runs an iterator
       case FOR_AWAIT_OF: // assigns to a loop LHS, runs an iterator, async operations.
+      case DYNAMIC_IMPORT:
         return true;
```

Comment:
missing `break;` here

Reply:
you want a `break;` even though there is a `return`?

Reply:
Sorry, I flagged the wrong spot.